File: backend/src/main/kotlin/dev/nazna/examplemonorepoapp/application/UserService.kt
package dev.nazna.examplemonorepoapp.application

import dev.nazna.examplemonorepoapp.domain.model.User
import dev.nazna.examplemonorepoapp.presentation.request.UserCreateRequest

interface UserService {

  fun findById(id: Long): User

  fun create(request: UserCreateRequest): User

}

File: sys-monorepo-webapp/aggregator/src/service/user.service.ts
import { Injectable } from '@nestjs/common'
import { User } from '../domain/model'

@Injectable()
export class UserService {
  async findById(id: string): Promise<User> {
    const result = new User({ id: `u-${id}`, name: 'John' })
    return result
  }
}
